### PR TITLE
virsh.migrate: enabled there_and_back migration

### PR DESCRIPTION
**Pull request by balamuruhans**
1. there_and_back migration was not enabled but cfg had the params for it.
2. pre migration setup and cleanup done using APIs

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
**_Tested with the PR:_**
>  (001/172) guest_migration.qemu.qcow2.scsi.smp2.virtio_net.CentOS.7.2.ppc64le.powerkvm-libvirt.virsh.migrate.there_and_back.normal_test:  PASS (161.51 s)
>  (002/172) guest_migration.qemu.qcow2.scsi.smp2.virtio_net.CentOS.7.2.ppc64le.powerkvm-libvirt.virsh.migrate.there_and_back.with_HP_only:  PASS (280.70 s)

**Comment by balamuruhans** [comment]
https://github.com/avocado-framework/avocado-vt/pull/804

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 1]
```diff
@@ -1066,6 +1120,8 @@ def check_migration_timeout_suspend(params):
     libvirt.setup_or_cleanup_nfs(False, export_dir=exp_dir,
                                  mount_dir=mount_dir,
                                  restore_selinux=local_selinux_bak)
+    # cleanup pre migration setup for remote machine
+    migrate_setup.migrate_pre_setup(dest_uri, params, cleanup=True)
```
Comment (sathnaga) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: Don't need this check  `if migrate_there_and_back:` while cleaning up?

Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: @sathnaga it is required for all migration test irrespective of there_and_back, as all the test cases perform migration from source to remote machine including there_and_back

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 2]
```diff
@@ -908,6 +916,52 @@ def check_migration_timeout_suspend(params):
                 raise error.TestError("%s did not respond after %d sec."
                                       % (vm.name, ping_timeout))
             server_session.close()
+            if migrate_there_and_back:
+                # pre migration setup for local machine
+                migrate_setup.migrate_pre_setup(src_uri, params)
+                logging.debug("Migration from %s to %s success" %
+                              (src_uri, dest_uri))
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: @sathnaga at this point migration from qemu:///system to remote machine is done.

**Comment by balamuruhans** [comment]
@waynesun09 @will-Do Please help me in reviewing this PR

**Comment by will-Do** [comment]
@dzhengfy
As you are working on migration part, please help to review this patch

**Comment by dzhengfy** [comment]
Ok, I will have a look at it. @will-Do

**Comment by balamuruhans** [comment]
@dzhengfy I have tested this PR and its working for me, if you dont have any comments let us merge this PR.

**Comment by dzhengfy** [comment]
@balamuruhans I have reviewed the changes and it looks good to me. Thanks.

**Comment by balamuruhans** [comment]
Thank you @will-Do @sathnaga @dzhengfy for the review. let us merge this

**Pull request merged**